### ISSUE TITLE
Question: What is yieldSteps?

### ISSUE THREAD

**Issue by jacj-simcorp** [hackwaly/ocamlearlybird#71]
The parameter `yieldSteps` is available in Launch configurations.
What is the purpose of this parameter?
What are reasonable bounds?
Are there any performance implications of setting this high or low?

Thanks for your time :)

**Comment by hackwaly** [comment]
The debuggee program's execution is threaded by the debugger. The `yieldSteps` is how many instructions the program can run in a batch. Because debugger and debuggee are run in same thread. So in a batch run: debugger are blocked and can't process any UI actions, such as set breakpoint, pause, etc.

The bigger `yieldSteps` result better performance. The smaller `yieldSteps` let you can have good opportunity to pause program at your desired position.

**Comment by jacj-simcorp** [comment]
> The debuggee program's execution is threaded by the debugger. The `yieldSteps` is how many instructions the program can run in a batch. Because debugger and debuggee are run in same thread. So in a batch run: debugger are blocked and can't process any UI actions, such as set breakpoint, pause, etc.
> 
> The bigger `yieldSteps` result better performance. The smaller `yieldSteps` let you can have good opportunity to pause program at your desired position.

Thanks.
When you say "pause program at your desired position", is this in relation to clicking "pause" or also for breakpoints? Or are breakpoints always hit, independently of the `yieldSteps` setting? I assume so?

**Comment by jacj-simcorp** [comment]
Also, with an extremely large setting, you are effectively then unable to "pause", I assume?

**Comment by hackwaly** [comment]
No, it does not affect to stepping. It only for setBreakpoint/pause a running program.
